site/en/articles/multi-screen-window-placement/index.md

```diff
@@ -143,6 +144,50 @@ try {
 }
 ```
 
+While browsers with the old and the new permission name are in use, be sure to use defensive code when requesting permission, as in the example below.
+
+```js
+document.querySelector('button').addEventListener('click', async () => {
+  let state;
+  let permissionName;
+  // The new permission name.
+  try {
+    state = (
+      await navigator.permissions.query({
+        name: 'window-management',
+      })
+    ).state;
+    permissionName = 'window-management'
+  } catch (err) {
+    if (err.name === 'TypeError') {
+      // The old permission name.
+      try {
+        state = (
+          await navigator.permissions.query({
+            name: 'window-placement',
+          })
+        ).state;
+        permissionName = 'window-placement'
+      } catch (err) {
+        if (err.name === 'TypeError') {
+          state = 'Window management not supported.';
+        } else {
+          state = `${err.name}: ${err.message}`;
+        }
+      }
+    } else {
+      state = `${err.name}: ${err.message}`;
+    }
+  }
+  if ('request' in Permissions.prototype) {
```

Comment:
This API has been behind the experimental web platform features flag for years. Are there signs this is going to ship soon?

Reply:
If not, my previous proposal could be reduced even further:

```js
async function getWindowManagementPermissionState() {
  let state;
  // The new permission name.
  try {
    ({ state } = await navigator.permissions.query({
      name: "window-management",
    }));
  } catch (err) {
    if (err.name !== "TypeError") {
      return `${err.name}: ${err.message}`;
    }
    // The old permission name.
    try {
      ({ state } = await navigator.permissions.query({
        name: "window-placement",
      }));
    } catch (err) {
      if (err.name === "TypeError") {
        return "Window management not supported.";
      }
      return `${err.name}: ${err.message}`;
    }
  }
  return state;
}

document.querySelector("button").addEventListener("click", async () => {
  const state = await getWindowManagementPermissionState();
  document.querySelector("pre").textContent = state;
});
```

Reply:
I don't think it's shipping anytime soon, even if I want it to. Removed. Thank you!